File: server/EvalutionAlghoritm/MinFitnessValue.js
import cloneDeep from "lodash/clonedeep.js";


export default function MinFitnessValue(populations, bestPopulation) {
  let min = Number.MAX_VALUE;
  let min_index;
  for (let i = 0; i < populations.length; i++) {
    if (populations[i].fitnessValue < min) {
      min = populations[i].fitnessValue;
      min_index = i;
    }
  }

  if (bestPopulation.fitnessValue > min) {
    return JSON.parse(JSON.stringify(populations[min_index]));
  }
  return bestPopulation;
}

File: server/EvalutionAlgoritmCpp/EvalutionAlgoritmCpp/EvulationAlgoritmCpp/MinFitnessValue.cpp
#include "MinFitnessValue.h"
individ MinFitnessValue(vector<individ>& populations, int start, int end)
{
	double min = INT_MAX;
	individ minIndivid;
	for(int i = start;i<end;i++)
	{
		if(populations[i].fitnessValue.fitnessValue<min)
		{
			minIndivid = populations[i];
			min = populations[i].fitnessValue.fitnessValue;
		}
	}
	
	return minIndivid;
}